### ISSUE TITLE
Training stops with no error after "AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n..." step

### ISSUE THREAD

**Issue by Ardibid** [ultralytics/ultralytics#5337]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Train

### Bug

(base) ubuntu@ip-172-31-77-117:~$ ! yolo task=detect mode=train model=yolov8s.pt data=/home/ubuntu/RoboFlow_pipeline/auto_prop_extended_dataset-3/data.yaml epochs=50 imgsz=800 batch=8 plots=True  device='cuda' 
Downloading https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov8s.pt to yolov8s.pt...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 21.5M/21.5M [00:00<00:00, 260MB/s]
New https://pypi.org/project/ultralytics/8.0.196 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.0.134 🚀 Python-3.10.6 torch-2.1.0+cu121 CUDA:0 (NVIDIA A10G, 22564MiB)
yolo/engine/trainer: task=detect, mode=train, model=yolov8s.pt, data=/home/ubuntu/RoboFlow_pipeline/auto_prop_extended_dataset-3/data.yaml, epochs=50, patience=50, batch=8, imgsz=800, save=True, save_period=-1, cache=False, device=cuda, workers=8, project=None, name=None, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, v5loader=False, tracker=botsort.yaml, save_dir=runs/detect/train
Overriding model.yaml nc=80 with nc=16

                   from  n    params  module                                       arguments                     
  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]                 
  1                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  2                  -1  1     29056  ultralytics.nn.modules.block.C2f             [64, 64, 1, True]             
  3                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  4                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  5                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  6                  -1  2    788480  ultralytics.nn.modules.block.C2f             [256, 256, 2, True]           
  7                  -1  1   1180672  ultralytics.nn.modules.conv.Conv             [256, 512, 3, 2]              
  8                  -1  1   1838080  ultralytics.nn.modules.block.C2f             [512, 512, 1, True]           
  9                  -1  1    656896  ultralytics.nn.modules.block.SPPF            [512, 512, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  1    591360  ultralytics.nn.modules.block.C2f             [768, 256, 1]                 
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 
 16                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 
 19                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  1   1969152  ultralytics.nn.modules.block.C2f             [768, 512, 1]                 
 22        [15, 18, 21]  1   2122240  ultralytics.nn.modules.head.Detect           [16, [128, 256, 512]]         
Model summary: 225 layers, 11141792 parameters, 11141776 gradients, 28.7 GFLOPs

Transferred 349/355 items from pretrained weights
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
Downloading https://github.com/ultralytics/assets/releases/download/v0.0.0/yolov8n.pt to yolov8n.pt...
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6.23M/6.23M [00:00<00:00, 195MB/s]
Segmentation fault (core dumped)

### Environment

ltralytics YOLOv8.0.134 🚀 
Python-3.10.6
torch-2.1.0+cu121 CUDA:0 (NVIDIA A10G, 22564MiB)
Setup complete ✅ (4 CPUs, 15.4 GB RAM, 66.5/124.0 GB disk)

### Additional

I am using RoboFlow CLI on top of YOLOv8

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by saikrishna0203** [comment]
Same doubt ,is that correct or wrong pls reply ,I have got amp checks passed with yolov8n model while I have yolov8s as command line. ? Pls say what to do

**Comment by glenn-jocher** [comment]
@saikrishna0203 hello 👋! It appears there is a bit of confusion regarding the AMP checks with different model variants in the training command.

The training command seems to be initiating correctly with `yolov8s.pt`. However, during the AMP checks, the output is indicating that it is downloading and using `yolov8n.pt`. This is unexpected behavior as the specified model in your command is `yolov8s.pt`.

This might be happening due to an internal decision in the code to compare AMP compatibility with a default model like `yolov8n.pt`, or it could be a glitch where the intended `yolov8s` model is not being used for the checks.

To better understand and resolve the issue, please double-check the following:

1. **Correct Model Usage**: Ensure that the model specified in your command (`yolov8s.pt`) is indeed the one you want to train with and that it's available or correctly downloaded in your environment.

2. **Updated Repository**: Make sure your YOLOv8 repository is up to date. There could have been updates or bug fixes since your last pull.

3. **AMP Compatibility**: It is possible that Automatic Mixed Precision (AMP) checks may run with a default lightweight model for compatibility testing before actual training starts. Make sure that this is not intended behavior by referring to the official documentation or updates in the repository.

4. **Training Configuration**: Confirm that all other training configurations are correct and that the dataset is correctly set up as per the requirements.

Considering the Segmentation fault (core dumped) error you've encountered, it's crucial to also ensure that your environment is correctly set up:

- **System libraries**: Check if there are any incompatible libraries that may need updating.
- **CUDA Compatibility**: Ensure your NVIDIA drivers, CUDA toolkit, and cuDNN libraries are compatible with the installed PyTorch version.
- **Hardware Resources**: Check if your system has sufficient resources and isn't running out of RAM or other resources during training.

If the issue persists, please visit the [Ultralytics Docs](https://docs.ultralytics.com) for more detailed guidance on AMP and training, and consider submitting an issue report on the YOLOv8 GitHub issues page with a detailed explanation of your environment and steps to reproduce the problem (following the guidelines for a minimum reproducible example).

Let us know how it goes or if you need further assistance! 🛠️
